File: 1732. Find the Highest Altitude/main.py
from typing import List


class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        altitude = 0
        largest = 0

        for val in gain:
            altitude += val
            largest = max(largest, altitude)

        return largest


class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        altitudes = [0]

        for i in range(1, len(gain) + 1):
            altitudes.append(altitudes[i - 1] + gain[i - 1])

        return max(altitudes)

File: 1732. Find the Highest Altitude/main.cpp
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    int largestAltitude(vector<int> &gain) {
        int altitude = 0;
        int largest = 0;

        for (const auto &val : gain) {
            altitude += val;
            largest = max(largest, altitude);
        }

        return largest;
    }
};

class Solution {
public:
    int largestAltitude(vector<int> &gain) {
        vector<int> altitudes(1, 0);

        for (int i = 1; i <= gain.size(); i++) {
            altitudes.push_back(altitudes[i - 1] + gain[i - 1]);
        }

        return *max_element(altitudes.begin(), altitudes.end());
    }
};